Src/Base/AMReX_MultiFabUtil_1D_C.H

```diff
@@ -198,7 +199,7 @@ void amrex_avgdown (Box const& bx, Array4<Real> const& crse,
     for (int n = 0; n < ncomp; ++n) {
         for (int i = clo.x; i <= chi.x; ++i) {
             int ii = i*facx;
-            Real c = 0.;
+            T c = 0.0*fine(ii,0,0,n+fcomp);
```

Comment:
@WeiqunZhang 
Would something like this be acceptable? (I'm guessing not...)
I want to see if there is a way to implement this without using a zero initializer, because there is not a consistent way to zero initialize for most data types. For instance if `T=amrex::Real` then you can do `T c = 0`, but if `T = eigen::Matrix3d` then you would need to initialize with `T c = T::Zero()`. 
Alternatively I could implement an SFINAE solution with a specialized version for `amrex::Real` that uses `T c = 0` and a general version for other types?

Reply:
I think it should be the opposite. `T c = 0` is certainly more general than `T::Zero`.  Maybe you can have a free function that returns `T(0)` and a specialized version for the types you want to use.  Ideally the function should be constexpr.